src/test/java/com/kodilla/ecommercee/domainTests/ProductTestSuite.java

```diff
@@ -0,0 +1,162 @@
+import com.kodilla.ecommercee.repository.OrderRepository;
+import com.kodilla.ecommercee.repository.ProductRepository;
+import jakarta.transaction.Transactional;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+@Transactional
+@SpringBootTest
+public class ProductTestSuite {
+    @Autowired
+    ProductRepository productRepository;
+    @Autowired
+    GroupRepository groupRepository;
+    @AfterEach
+    public void cleanup() {
+        productRepository.deleteAll();
+        groupRepository.deleteAll();
+    }
+
+    @Test
+    public void testAddProductToDatabase() {
+    // Given
+        Group group = new Group();
+        group.setName("Test group");
+        group.setDescription("Test Group Description");
+        Group savedGroup = groupRepository.save(group);
+        Product product = new Product();
+        product.setName("Test Product");
+        product.setDescription("Test Description");
+        product.setPrice(BigDecimal.valueOf(10.00));
+        product.setGroup(savedGroup);
+        product.setCarts(Arrays.asList(new Cart()));
+        product.setOrders(Arrays.asList(new Order()));
+        product.setActive(true);
+
+    //When
+     Product savedProduct = productRepository.save(product);
+    //Then
+        assertNotNull(savedGroup);
+        assertEquals("Test Product", savedProduct.getName());
+        assertEquals("Test Description", savedProduct.getDescription());
+        assertEquals(BigDecimal.valueOf(10.00), savedProduct.getPrice());
+        assertNotNull(savedProduct.getGroup());
+        assertEquals(1, savedProduct.getCarts().size());
+        assertEquals(1, savedProduct.getOrders().size());
+        assertTrue(savedProduct.isActive());
+    //CleanUp
+        productRepository.deleteAll();
+        groupRepository.deleteAll();
+    // Verify cleanup
+        assertEquals(0, productRepository.findAll().size());
+        assertEquals(0, groupRepository.findAll().size());}
+    @Test
+    public void testFindProductById() {
+    // Given
+        Product product = new Product();
+        product.setName("Test Product");
+        product.setDescription("Test Description");
+        product.setPrice(BigDecimal.valueOf(10.00));
+        product.setActive(true);
+        Product savedProduct = productRepository.save(product);
+
+    // When
+        Long productId = savedProduct.getId();
+        Product retrievedProduct = productRepository.findById(productId).orElse(null);
+
+    // Then
+        assertNotNull(retrievedProduct);
+        assertEquals(productId, retrievedProduct.getId());
```

Comment:
wydaje mi się, że powinno być porównanie id do "product", bo w tym co teraz jest porównywane jest to samo do siebie, czyli produkt z repozytorium do produkt z repozytorium, zamiast do product

```suggestion
        productRepository.save(product);

    // When
        Product retrievedProduct = productRepository.findById(product.getId()).orElse(null);

    // Then
        assertNotNull(retrievedProduct);
        assertEquals(product.getId(), retrievedProduct.getId());
```